docs/README.md

```diff
@@ -658,22 +658,244 @@ In this chapter you'll walk through these concepts and its details.
+
+Additionally, commands must be within the commands directory of the project source: `project-root/src/commands`.
+
+```text
+project-root
+├── src
+│   ├── commands <------ They must be here
+│   ├── common
+│   ├── config
+│   ├── entities
+│   ├── events
+│   ├── index.ts
+│   └── read-models
+```
+
 #### Creating a command
 
+The preferred way to create a command is by using the generator, e.g.
+
+```shell
+boost new:command CreateProduct --fields sku:SKU displayName:string description:string price:Money
+```
+
+The generator will automatically create a file called `CreateProduct.ts` with a TypeScript class of the same name under the commands directory. You can still create the command manually. Since the generator is not doing any *magic*, all you need is a class decorated as `@Command`. Anyway, we recommend you always to use the generator, because it handles the boilerplate code for you.
+
+Note:
+> Running the command generator with a `CommandName` already existing, will override the content of the current one. Soon, we will display a warning before overwriting anything. Meantime, if you missed a field, just add it to the class because in Booster, all the infrastructure and data structures are inferred from your code.
+
 #### The command handler function
 
+Booster generates the command handler function as a method of the command class. That function is called by the framework every time that one instance of this command is submitted. This is the right place to run validations, return errors, query entities to make decisions and register relevant domain events.
+
 ##### Validating data
 
+Booster uses the typed nature of GraphQL to ensure that types are correct before reaching the handler, so you don't have to validate types.
+
+###### Throw an error
+
+There are still business rules to be checked before proceeding with a command. For example, a given number must be between a threshold or a string must match a regular expression. In that case, it is enough just to throw an error in the handler, and then Booster will use the error's message as the response to make it descriptive, e.g.
+
+Given this command:
+
+```typescript
+@Command({
+  authorize: 'all',
+})
+export class CreateProduct {
+  public constructor(
+    readonly sku: string,
+    readonly price: number
+  ) {}
+
+  public async handle(register: Register): Promise<void> {
+    const priceLimit = 10
+    if (this.price >= priceLimit) {
+      throw new Error(`price must be below ${priceLimit}, and it was ${this.price}`)
+    }
+  }
+}
+
+```
+
+And this mutation:
+
+```graphql
+mutation($input: CreateProductInput!) {
+  CreateProduct(input: $input)
+}
+
+# Variables
+
+{
+  "input": {
+    "sku": "MYSKU",
+    "price": 19.99
+  }
+}
+```
+
+You'll get something like this response:
+
+```grapqhl
+{
+  "errors": [
+    {
+      "message": "price must be below 10, and it was 19.99",
+      "path": [
+        "CreateProduct"
+      ]
+    }
+  ]
+}
+```
```

Comment:
That moment when you didn't know about this and yet you coded part of the exception handling 😂